src/utilities/helpers.ts

```diff
@@ -16,3 +24,31 @@ export const formatTime = (seconds: number) => {
   let secondsFormatted = seconds % 60 > 9 ? seconds % 60 : "0" + (seconds % 60);
   return hoursFormatted + minutesFormatted + secondsFormatted;
 };
+
+const setStreak = async (count: number) => {
+  await AsyncStorage.setItem(
+    "@streak",
+    JSON.stringify({ count, timestamp: moment().toISOString() }),
+  );
+  return count;
+};
+
+export const updateStreak = async (): Promise<number> => {
+  const streakData = await AsyncStorage.getItem("@streak").catch((e) =>
+    console.log(e),
+  );
+  if (streakData) {
+    const { count, timestamp } = JSON.parse(streakData) as Streak;
+    const lastMeditated = moment(timestamp);
+    const today = moment();
+    if (
+      !today.isSame(lastMeditated, "day") &&
+      today.diff(lastMeditated, "hours") < 24
+    ) {
+      return setStreak(count + 1);
+    } else if (today.isSame(lastMeditated, "day")) {
+      return setStreak(count);
```

Comment:
Hmm, do we need this call? Seems like we could just pass

Reply:
??

Reply:
ah see what you mean

Reply:
oh, no I don't

Reply:
Why call `setStreak(count)` if the last meditated day in the system is today?

The count and last day meditated variables should already be correctly stored, so it is an unnecessary to store them again.

Reply:
Yeah you could just return I guess